diff --git a/include/private/gcconfig.h b/include/private/gcconfig.h
index 6970e6f2..a3af3824 100644
--- a/include/private/gcconfig.h
+++ b/include/private/gcconfig.h
@@ -2480,6 +2480,8 @@ EXTERN_C_BEGIN
 #     endif
 #     undef USE_MMAP /* similar to Emscripten */
 #     undef USE_MUNMAP
+      /* The real page size in WebAssembly is 64 KB.    */
+#     define GETPAGESIZE() 65536
 #     if defined(GC_THREADS) && !defined(CPPCHECK)
 #       error No threads support yet
 #     endif
